Misc/NEWS.d/next/Core and Builtins/2024-05-23-20-17-37.gh-issue-119258.wZFIpt.rst

```diff
@@ -0,0 +1,3 @@
+Eliminate type version guards in the tier two interpreter.
+
+Note that setting the ``tp_version_tag`` manually (which has never been supported) may result in crashes.
```

Comment:
There's a test that sets it manually in `test/test_type_cache.py`. Could you decorate that with `without_optimizer` from `test.support` please?

Reply:
I believe this test has already been updated to use the new API function, which updates the cache correctly.

Reply:
Yeah I believe that is correct, `type_assign_specific_version_unsafe` was moved to `_testinternalcapi` since it now uses an internal API and the tests were updated to use it.